admin/nodes/views.py

```diff
@@ -228,6 +233,34 @@ def get_context_data(self, **kwargs):
             'page': page,
         }
 
+class RegistrationUpdateEmbargoView(PermissionRequiredMixin, View):
+    """ Allow authorized admin user to update the embargo of a registration
+    """
+    permission_required = ('osf.change_node')
+    raise_exception = True
+
+    def post(self, request, *args, **kwargs):
+        end_date = request.POST.get('date')
+        user = request.user
+
+        try:
+            end_date = pytz.utc.localize(datetime.strptime(end_date, '%m/%d/%Y'))
+        except ValueError:
+            return HttpResponse('Please enter a valid date.', status=400)
+
+        try:
+            registration = self.get_object()
+            change_embargo_date(registration, user, end_date)
+        except ValidationError as e:
+            return HttpResponse(e, status=409)
+        except PermissionDenied as e:
+            return HttpResponse(e, status=403)
+
+        return redirect(reverse_node(self.kwargs.get('guid')))
+
+    def get_object(self, queryset=None):
+        return Registration.load(self.kwargs.get('guid'))
```

Comment:
`Registration.objects.get(_id=self.kwargs.get('guid'))`
`load` is old syntax. It still exists but I think all refs have been removed for the most part. (probably for us to be lazy when using shell)